moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp

```diff
@@ -354,32 +355,33 @@ bool RuckigSmoothing::runRuckig(robot_trajectory::RobotTrajectory& trajectory,
         duration_extension_factor *= DURATION_EXTENSION_FRACTION;
         // Reset the trajectory
         trajectory = robot_trajectory::RobotTrajectory(original_trajectory, true /* deep copy */);
-        for (size_t time_stretch_idx = 1; time_stretch_idx < num_waypoints; ++time_stretch_idx)
+
+        size_t time_stretch_idx = waypoint_idx;
+        trajectory.setWayPointDurationFromPrevious(
+            time_stretch_idx + 1,
+            duration_extension_factor * original_trajectory.getWayPointDurationFromPrevious(time_stretch_idx + 1));
+        // re-calculate waypoint velocity and acceleration
+        auto target_state = trajectory.getWayPointPtr(time_stretch_idx + 1);
+        const auto prev_state = trajectory.getWayPointPtr(time_stretch_idx);
+        timestep = trajectory.getAverageSegmentDuration();
```

Comment:
I'm fairly sure this line is wrong. We don't want AverageSegmentDuration -- we want this duration that you calculated at L360:

`timestep = getWayPointDurationFromPrevious(time_stretch_idx + 1);`